.github/workflows/ci.protocol.yaml

```diff
@@ -0,0 +1,34 @@
+name: Continuous Integration - Protocol
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - 'protocol/**'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Protocol Tests
+        uses: actions/setup-node@v1
+        with:
+          node-version: '16'
+      - name: Cache Node Modules
+        id: node-modules-cache
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-modules-
+      - name: Install Dependencies
+        if: steps.node-modules-cache.outputs.cache-hit != 'true'
+        run: yarn install --immutable
+      - run: yarn compile
```

Comment:
you don't need to run compile, generate does that I believe

Reply:
https://github.com/BeanstalkFarms/Beanstalk/pull/255/commits/b3027157b374bf80e3be5a381e3a5bf5849717c2